vtex-apps/order-coupon
# Remove coupon info from OrderCoupon

#### What problem is this solving?

This PR removes from `OrderCoupon` the responsibility of providing the selected `coupon` value. With this, we ensure that `OrderManager` is the only source of truth when it comes to `orderForm` data, and we make `OrderCoupon` concentrate only manipulation functions.



#### How should this be manually tested?

- Proceed to [Workspace](https://ordercoupontests--checkoutio.myvtex.com/cart)
- Try to insert the coupon `testegratis`
- Check if it works properly

#### Checklist/Reminders

- [x] [Linked this PR to a Clubhouse story (if applicable).](https://app.clubhouse.io/vtex/story/29728/remover-informa%C3%A7%C3%A3o-de-coupon-de-ordercoupon-transformando-o-orderform-na-%C3%BAnica-fonte-de-dados)

#### Screenshots or example usage

#### Type of changes


✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
✔️| Breaking change 
_ | Technical improvements
